scripts/inject-metadata

```diff
@@ -0,0 +1,68 @@
+#!/usr/bin/env python3
+
+"""
+Accept a template schema on stdin and inject default metadata fields
+before writing to stdout.
+
+Expects the target schema path to be passed as a command-line argument.
+"""
+
+import argparse
+from collections.abc import Mapping
+import json
+from pathlib import Path
+import sys
+
+MOZ_PIPELINE_METADATA = "mozPipelineMetadata"
+TEMPLATES_DIR = Path(__file__).parent.parent / "templates"
+
+
+def merge_where_empty(a, b):
+    """Recursively merge b into a where items are missing in a."""
+    for key, b_value in b.items():
+        a_value = a.get(key)
+        if a_value is None:
+            a[key] = b_value
+        elif isinstance(a_value, Mapping) and isinstance(b_value, Mapping):
+            merge_where_empty(a_value, b_value)
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser(description=__doc__)
+    parser.add_argument("target")
+    args = parser.parse_args()
+
+    fileparts = Path(args.target).parts
+    schema_basename = fileparts[-1]
+    doctype = fileparts[-2]
+    namespace = fileparts[-3]
+
+    # Parse the schema and extract metadata.
+    schema = json.load(sys.stdin)
+    meta = schema.get(MOZ_PIPELINE_METADATA, {})
+
+    # Namespace-level defaults.
+    defaults_path = TEMPLATES_DIR / namespace / "defaults.schema.json"
```

Comment:
I'm not sure that namespace-level defaults belong in `templates` either. Since these shouldn't need the capabilities of generating the schemas, we should move them to a separate directory.